src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs

```diff
@@ -301,25 +302,26 @@ private void PopulateHtmlFile()
                 Environment.GetEnvironmentVariable("UserName"), Environment.MachineName,
                 FormatDateTimeForRunName(DateTime.Now));
 
-            XmlFilePath = GetFilePath(HtmlLoggerConstants.XmlFileExtension, fileName);
+            XmlFilePath = GenerateUniqueFilePath(fileName, HtmlLoggerConstants.XmlFileExtension);
 
-            using (var xmlStream = _fileHelper.GetStream(XmlFilePath, FileMode.Create))
+            using (var xmlStream = _fileHelper.GetStream(XmlFilePath, FileMode.OpenOrCreate))
             {
                 _xmlSerializer.WriteObject(xmlStream, TestRunDetails);
             }
 
-            if (string.IsNullOrEmpty(HtmlFilePath))
-            {
-                HtmlFilePath = GetFilePath(HtmlLoggerConstants.HtmlFileExtension, fileName);
-            }
+            HtmlFilePath = GenerateUniqueFilePath(
+                string.IsNullOrEmpty(HtmlFilePath)
+                    ? fileName
+                    : Path.GetFileNameWithoutExtension(HtmlFilePath),
```

Comment:
I am assuming that when reusing the already provided name we could have duplicate so I want to generate a unique name out of the given name.